code/modules/power/cable.dm

```diff
@@ -87,41 +80,53 @@ GLOBAL_LIST_INIT(wire_node_generating_types, typecacheof(list(/obj/structure/gri
 					if(term && (!term.master || term.master == search_parent))
 						continue
 				if(UNDER_TERMINAL)
-					var/obj/machinery/power/smes/S = locate(/obj/machinery/power/smes) in TB
-					if(S && (!S.terminal || S.terminal == search_parent))
+					var/obj/machinery/power/smes/smes = locate(/obj/machinery/power/smes) in step_turf
+					if(smes && (!smes.terminal || smes.terminal == search_parent))
 						continue
+
 		var/inverse = turn(check_dir, 180)
-		for(var/obj/structure/cable/C in TB)
-			if(C.cable_layer & cable_layer)
+
+		for(var/obj/structure/cable/other_cable in step_turf.nullspaced_contents)
+			if(other_cable.cable_layer & cable_layer)
 				linked_dirs |= check_dir
-				C.linked_dirs |= inverse
-				C.update_appearance()
+				other_cable.linked_dirs |= inverse
+				other_cable.update_appearance()
 
 	update_appearance()
 
 ///Clear the linked indicator bitflags
 /obj/structure/cable/proc/Disconnect_cable()
 	for(var/check_dir in GLOB.cardinals)
 		var/inverse = turn(check_dir, 180)
-		if(linked_dirs & check_dir)
-			var/TB = get_step(loc, check_dir)
-			for(var/obj/structure/cable/C in TB)
-				if(cable_layer & C.cable_layer)
-					C.linked_dirs &= ~inverse
-					C.update_appearance()
+		if(!(linked_dirs & check_dir))
+			continue
+
+		var/turf/neighbor_turf = get_step(associated_loc, check_dir)
+
+		for(var/obj/structure/cable/neighbor_cable in neighbor_turf.nullspaced_contents)
+			if(cable_layer & neighbor_cable.cable_layer)
+				neighbor_cable.linked_dirs &= ~inverse
+				neighbor_cable.update_appearance()
 
 /obj/structure/cable/Destroy() // called when a cable is deleted
+	if(associated_loc && !loc)
+		loc = associated_loc
```

Comment:
what's the deal with this? don't think it was in pipe code
is there a reason for this stuff to not be centralized?

Reply:
no i think it can safely be handled by the component